src/hpc/threadapi.c

```diff
@@ -2042,8 +2046,13 @@ MigrateObjects(int count, Obj * objects, Region * target, int retype)
                 return 0;
         }
     }
-    for (i = 0; i < count; i++)
-        SET_REGION(objects[i], target);
+    for (i = 0; i < count; i++) {
+        Obj obj = objects[i];
+        if (TNUM_OBJ(obj) == T_PREC) {
+            SortPRecRNam(obj, 0);
```

Comment:
Please add a comment that explain what is going on here.

Reply:
Added, this is about unsorted records being sorted if they are moved to a region where they cannot be mutated (or only if you have write access).